Add troubleshooting for intermittently missing Step Functions traces (#24400)
DataDog/documentation: content/en/serverless/step_functions/troubleshooting.md

## I can see the `aws.stepfunctions` root span but I cannot see any step spans
Please enable the `Include execution data` option on the state machine's logging. After enabling this option, log execution input, data passed between states, and execution output is logged. The Datadog backend uses the logs to construct these step spans for you.

## Traces are missing intermittently

> When searching traces, select the **Live Search** option on the upper right corner. If Live Search shows your trace, add "@trace_type:stepfunctions" to the [retention filter](https://docs.datadoghq.com/tracing/trace_pipeline/trace_retention/#retention-filters) and set the desired retention rate. For debugging, we recommend setting the retention rate to 100%. The filter can be disabled after debugging is done.

Reviewer: ```suggestion
When searching traces, select the **Live Search** option in the upper right corner. If Live Search shows your trace, add "@trace_type:stepfunctions" to the [retention filter](https://docs.datadoghq.com/tracing/trace_pipeline/trace_retention/#retention-filters) and set the desired retention rate. For debugging, Datadog recommends setting the retention rate to 100%. The filter can be disabled after debugging is done.
```
